### PR TITLE
🐛 Fix upgrading mariadb container in upgrade ironic e2e tests

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
After the separation of mariadb-image from ironic-image in ironic deployment ([332a5](https://github.com/metal3-io/baremetal-operator/pull/1052/commits/332a5ca6ccce2167236cd8ba7d92d834c2057208)), E2E upgrading ironic containers test is failing due to the old configuration, where all containers were pointing to the same image (`image: quay.io/metal3-io/ironic`). Since now mariadb container has its own image (`image: quay.io/metal3-io/mariadb`), this PR fixes that by differentiating the containers and pointing to proper image based on that. 

Also introduces minor changes:
- adds more logs to upgrade ironic container tests to keep consistent with other e2e tests (starting && ending)
- switches the order of cert rotation e2e tests since main branch tests it after ironic upgrade tests.

### CODE REVIEW

File: test/e2e/upgrade_ironic_test.go [hunk 1]
```diff
@@ -28,16 +29,11 @@ func upgradeIronic() {
...
-			"mariadb",
-			"ironic-api",
-			"ironic-dnsmasq",
-			"ironic-conductor",
-			"ironic-log-watch",
-			"ironic-inspector":
-			deploy.Spec.Template.Spec.Containers[i].Image = containerRegistry + "/metal3-io/ironic:" + ironicImageTag
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == "mariadb" {
+			container.Image = containerRegistry + "/metal3-io/mariadb:" + ironicImageTag
```
Comment (mboukhalfa) [inline test/e2e/upgrade_ironic_test.go:34]: Probably better to create a specific var for mariadb just like 
```suggestion
                        mariadbImageTag := ironicImageTag // or in e2eConfig
			container.Image = containerRegistry + "/metal3-io/mariadb:" + mariadbImageTag
```

**Comment by furkatgofurov7** [comment]
/retitle 🐛 Fix upgrading mariadb container in upgrade ironic e2e tests
/test-v1a5-e2e
/test-v1a5-centos-e2e
/test-v1a5-integration

**Comment by mboukhalfa** [comment]
LGTM

**Review by kashifest — APPROVED** [review]

**Review by namnx228 — COMMENTED** [review]
/lgtm

**Pull request merged**